Sources/ApplicativeRouter/SyntaxRouter.swift

```diff
@@ -0,0 +1,159 @@
+import Foundation
+import Prelude
+import Optics
+
+// MARK: - Syntax Router
+
+// TODO: should this be generic over any monoid `M` instead of using `RequestData` directly?
+// TODO: generic over an error semigroup too?
+public struct Router<A> {
+  let parse: (RequestData) -> (rest: RequestData, match: A)?
+  let print: (A) -> RequestData?
+  let template: (A) -> RequestData?
```

Comment:
the router simultaneously holds the functions that will parse, print and document routes.